chart/templates/controller.yaml

```diff
@@ -0,0 +1,202 @@
+kind: ClusterRoleBinding
+metadata:
+  name: '{{.Values.wds_cp_name}}-otp-manager-rolebinding'
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: '{{.Values.wds_cp_name}}-otp-manager-role'
+subjects:
+  - kind: ServiceAccount
+    name: transport-controller-sa
+    namespace: '{{.Values.wds_cp_name}}-system'
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: transport-controller-config
+  namespace: {{.Values.wds_cp_name}}-system
+data:
+  get-kubeconfig.sh: |
+    #!/bin/bash
+    # this script receives a ControlPlane name and a parameter
+    # that determines whether to extract the ControlPlane's in-cluster kubeconfig
+    # or the external kubeconfig (if set to "true", the first will be retrieved).
+    # The function returns the requested kubeconfig's content in base64.
+    # it assumes the kubeconfig context is set to access the hosting cluster.
+
+    cpname="$1"
+    get_incluster_key="$2"
+
+    key=""
+    if [[ "$get_incluster_key" == "true" ]]; then
+      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.inClusterKey}');
+    else
+      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.key}');
+    fi
+
+    # get secret details
+    secret_name=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.name}')
+    secret_namespace=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.namespace}')
+
+    # get the kubeconfig in base64
+    kubectl get secret $secret_name -n $secret_namespace -o=jsonpath="{.data.$key}"
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: transport-controller
+  namespace: {{.Values.wds_cp_name}}-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: transport-controller
+  template:
+    metadata:
+      labels:
+        name: transport-controller
+    spec:
+      serviceAccountName: transport-controller-sa
+      initContainers:
+      - name: setup-wds-kubeconfig
+        image: quay.io/kubestellar/kubectl:1.27.8
+        imagePullPolicy: Always
+        command: [ "bin/sh", "-c", "sh /mnt/config/get-kubeconfig.sh {{.Values.wds_cp_name}} true | base64 -d > /mnt/shared/wds-kubeconfig"]
+        volumeMounts:
+        - name: config-volume
+          mountPath: /mnt/config
+        - name: shared-volume
+          mountPath: /mnt/shared
+      - name: setup-imbs-kubeconfig
+        image: quay.io/kubestellar/kubectl:1.27.8
+        imagePullPolicy: Always
+        command: [ "bin/sh", "-c", "sh /mnt/config/get-kubeconfig.sh {{.Values.transport_cp_name}} true | base64 -d > /mnt/shared/transport-kubeconfig"]
+        volumeMounts:
+        - name: config-volume
+          mountPath: /mnt/config
+        - name: shared-volume
+          mountPath: /mnt/shared
+      containers:
+        - name: transport-controller
+          # image: OTP_IMAGE_PLACEHOLDER
+          image: ghcr.io/kubestellar/ocm-transport-plugin/transport-controller:latest

```

Comment:
is this a typo?
you meant to put here `image: OTP_IMAGE_PLACEHOLDER`?
if yes, please uncomment and remove the other line with `image`.

Reply:
sorry, again, I pushed after testing 
I was short on time, I should have double checked